gems/smithy-client/lib/smithy-client/protocols/rpcv2.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Protocols
+      # RPCV2 protocol
+      class RPCV2 < ProtocolBase
+        def initialize
+          super('smithy.protocols#rpcv2Cbor')
+        end
+
+        def build_request(context)
+          puts 'BUILDING REQUEST FROM RPCV2 PROTOCOL'
+          input_shape = context.operation.input
+          context.request.http_method = 'POST'
+          codec = Client::Codecs::CBOR.new(codec_setting(input_shape))
+          context.request.body = codec.serialize(context.params, input_shape)
+          apply_headers(context.request)
+        end
+
+        def parse_response(context)
+          puts 'PARSING RESPONSE FROM RPCV2 PROTOCOL'
+          # TODO: Need to refactor to handle eventstreams
+          output_shape = context.operation.output
+          codec = Client::Codecs::CBOR.new(codec_setting(output_shape))
+          codec.deserialize(context.response.body, output_shape, output_shape.type)
+        end
+
+        # TODO: Implement ContentType Handler specific to RPCV2
```

Comment:
Downsides of the idea behind "codegenerated plugins" imo, less control over protocol-specific handlers.  I haven't quite figured this part out that works with codegenerated plugin for protocol.

Maybe its better to not have codegenerated plugin.... still thinking things through but open to ideas.